### PR TITLE
Add OPA support to the provider

### PR DESCRIPTION
**Pull request by mrinalirao**
## Description

Spike to add OPA support to the tfe provider. 

## Testing plan

Acceptance tests added

## External links

- [go-tfe documentation]()
- [API documentation](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/policies)

### CODE REVIEW

File: tfe/resource_tfe_policy.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+			},
+
+			"policy": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"enforce_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (glennsarti) [inline tfe/resource_tfe_policy.go:62]: Probably worth adding a note here which enforcements are valid for which policy kind.

File: tfe/resource_tfe_policy.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+			"organization": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"kind": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "sentinel",
```
Comment (glennsarti) [inline tfe/resource_tfe_policy.go:47]: Do we expect an enum here?

File: tfe/resource_tfe_policy.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
```
Comment (glennsarti) [inline tfe/resource_tfe_policy.go:35]: This isn't computed is it?

File: tfe/resource_tfe_policy.go [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+						string(tfe.EnforcementMandatory),
+					},
+					false,
+				),
+			},
+		},
+	}
+}
+
+func resourceTFEPolicyCreate(d *schema.ResourceData, meta interface{}) error {
```
Comment (glennsarti) [inline tfe/resource_tfe_policy.go:77]: I wonder if it's worth using "creator" funcs here for each type of policy e.g.

``` golang
func resourceTFEPolicyCreate(d *schema.ResourceData, meta interface{}) error {
   ...
  // Setup common policy options
  options := tfe.PolicyCreateOptions{}
  //  Setup per-kind policy options
  switch vKind {
  case Kind::OPA:
    options, err = createOPAPolicyOptions(options, d, meta)
  case Kind::Sentinel:
    options, err = createSentinelPolicyOptions(options, d, meta)
  default:
    return <raise some error>
  }
  if err != nil { return err }
  ....
}

File: tfe/resource_tfe_policy_set.go [hunk 5]
```diff
@@ -46,6 +46,17 @@ func resourceTFEPolicySet() *schema.Resource {
...
 
+			"kind": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "sentinel",
+			},
+
+			"overridable": {
+				Type:     schema.TypeBool,
+				Optional: true,
```
Comment (glennsarti) [inline tfe/resource_tfe_policy_set.go:57]: Is there a default here?

File: go.mod [hunk 6]
```diff
@@ -66,3 +66,5 @@ require (
 	github.com/vmihailenco/msgpack/v4 v4.3.12 // indirect
 	github.com/vmihailenco/tagparser v0.1.1 // indirect
 )
+
+replace github.com/hashicorp/go-tfe => ../go-tfe
```
Comment (glennsarti) [inline go.mod:70]: might be worth adding a `// TODO DEBUG ONLY. REMOVE ME` comment above this :-)

File: tfe/provider.go [hunk 7]
```diff
@@ -105,6 +105,7 @@ func Provider() *schema.Provider {
 			"tfe_organization_module_sharing": resourceTFEOrganizationModuleSharing(),
 			"tfe_organization_run_task":       resourceTFEOrganizationRunTask(),
 			"tfe_organization_token":          resourceTFEOrganizationToken(),
+			"tfe_policy":                      resourceTFEPolicy(),
```
[the remark this answers is not in the document]
Comment (cam-stitt) [reply tfe/provider.go:108]: They are simply a "policy", we just use the term "managed policies" to cite the difference between a policy that has come via a VCS connection and a policy that is individually managed. I'm happy with `tfe_policy`

**Comment by mrinalirao** [comment]
Closing this spike in favour of creating two separate, smaller PRs for review

**Pull request closed without merging**